test/downstream/integrator_indexing.jl

```diff
@@ -37,111 +37,100 @@ p = [lorenz1.σ => 10.0,
 # Check if indexing using variable names from interpolated integrator works
-interpolated_integrator = integrator(0.0:1.0:10.0)
-@test interpolated_integrator[α] isa Vector
-@test interpolated_integrator[α, :] isa Vector
-@test interpolated_integrator[α, 2] isa Float64
-@test length(interpolated_integrator[α, 1:5]) == 5
-@test interpolated_integrator[α] ≈
-      2interpolated_integrator[lorenz1.x] .+ interpolated_integrator[a] .* 2.0
-@test collect(interpolated_integrator[t]) isa Vector
-@test collect(interpolated_integrator[t, :]) isa Vector
-@test interpolated_integrator[t, 2] isa Float64
-@test length(interpolated_integrator[t, 1:5]) == 5
-
-integrator1 = integrator(0.0:1.0:10.0)
-@test integrator1.u isa Vector
-@test first(integrator1.u) isa Vector
-@test length(integrator1.u) == 11
-@test length(integrator1.t) == 11
-
-integrator2 = integrator(0.1)
-@test integrator2 isa Vector
-@test length(integrator2) == length(states(sys_simplified))
-@test first(integrator2) isa Real
-
-integrator3 = integrator(0.0:1.0:10.0, idxs = [lorenz1.x, lorenz2.x])
-@test integrator3.u isa Vector
-@test first(integrator3.u) isa Vector
-@test length(integrator3.u) == 11
-@test length(integrator3.t) == 11
-@test collect(integrator3[t]) ≈ integrator3.t
-@test collect(integrator3[t, 1:5]) ≈ integrator3.t[1:5]
-@test integrator(0.0:1.0:10.0, idxs = [lorenz1.x, 1]) isa RecursiveArrayTools.DiffEqArray
-
-integrator4 = integrator(0.1, idxs = [lorenz1.x, lorenz2.x])
-@test integrator4 isa Vector
-@test length(integrator4) == 2
-@test first(integrator4) isa Real
-@test integrator(0.1, idxs = [lorenz1.x, 1]) isa Vector{Real}
-
-integrator5 = integrator(0.0:1.0:10.0, idxs = lorenz1.x)
-@test integrator5.u isa Vector
-@test first(integrator5.u) isa Real
-@test length(integrator5.u) == 11
-@test length(integrator5.t) == 11
-@test collect(integrator5[t]) ≈ integrator3.t
-@test collect(integrator5[t, 1:5]) ≈ integrator3.t[1:5]
-@test_throws Any integrator(0.0:1.0:10.0, idxs = 1.2)
-
-integrator6 = integrator(0.1, idxs = lorenz1.x)
-@test integrator6 isa Real
-@test_throws Any integrator(0.1, idxs = 1.2)
-
-integrator7 = integrator(0.0:1.0:10.0, idxs = [2, 1])
-@test integrator7.u isa Vector
-@test first(integrator7.u) isa Vector
-@test length(integrator7.u) == 11
-@test length(integrator7.t) == 11
-@test collect(integrator7[t]) ≈ integrator3.t
-@test collect(integrator7[t, 1:5]) ≈ integrator3.t[1:5]
-
-integrator8 = integrator(0.1, idxs = [2, 1])
-@test integrator8 isa Vector
-@test length(integrator8) == 2
-@test first(integrator8) isa Real
-
-integrator9 = integrator(0.0:1.0:10.0, idxs = 2)
-@test integrator9.u isa Vector
-@test first(integrator9.u) isa Real
-@test length(integrator9.u) == 11
-@test length(integrator9.t) == 11
-@test collect(integrator9[t]) ≈ integrator3.t
-@test collect(integrator9[t, 1:5]) ≈ integrator3.t[1:5]
-
-integrator10 = integrator(0.1, idxs = 2)
-@test integrator10 isa Real
+# It doesn't because this returns a Vector{Vector{T}} and not a DiffEqArray
```

Comment:
Good point, that would be good to solve separately.

Reply:
Open an issue?

Reply:
Sure